### DISCUSSION TITLE
Example helloworld for esp32 says "Expected ESP-IDF version v4.3-beta2."

### DISCUSSION THREAD

**Discussion by dumdiedum** [Moddable-OpenSource/moddable#945] [Getting started] [answered]
Hello there! 

I'm new here, so maybe I missed something while reading instructions for installing Moddable SDK and ESP-IDF v4.4.

My esp32 is connected on port COM4, so I set an environment variable "UPLOAD_PORT"

The helloword examples runs well on platform "win", but when I try to run it on an esp32 platform I get following error:

<path>\moddable\build\tmp\esp32\nodemcu\debug\helloworld\makefile(190) : fatal error U1050: Detected ESP-IDF version v4.4.2. Expected ESP-IDF version v4.3-beta2.
Stop.

I've installed ESP-IDF 4.3 but same error appeard. Do anybody has an idea what I'm missing?

**Comment by andycarle** [comment]
Greetings! 

I'm sorry that you're having trouble building for the ESP32. Let's see if we can get you back up and running here.

The error message that you've included is a little curious, as it's the opposite of what I would expect it to be. The Moddable SDK build currently expects ESP-IDF version v4.4.2, as can be seen in the [ESP32 makefile fragment for Windows](https://github.com/Moddable-OpenSource/moddable/blob/4fded727f3cdc77b47ceca88a0040848070e98be/tools/mcconfig/nmake.esp32.mk#L23). That error also does not quite match what we output on an ESP-IDF version mismatch. Is there any chance that error got mangled a bit in moving it over to GitHub?

Either way, the right fix should be to get your ESP-IDF repository onto the `v4.4.2` branch. You can do this with commands like:

```console
cd %IDF_PATH%
git fetch --all --tags
git checkout v4.4.2
git submodule update --init --recursive
```

After doing that, if you run the command `git describe --always --abbrev=0` you should see the output `v4.4.2`. 

Please give that a try and let me know how it goes!

**Comment by andycarle** [comment, marked as answer]
Oh! Is there any chance that you have a very out-of-date Moddable SDK somehow? Or that you have not rebuilt the Moddable SDK since updating? That would explain the inconsistencies. 

If there's any chance you have an old version of the Moddable SDK, please try working through the [updating Windows instructions](https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/Moddable%20SDK%20-%20Getting%20Started.md#win-update) in our Getting Started guide.

**Comment by dumdiedum** [comment]
Oh snap, I didn't pay attention to the checked out branch of Moddable SDK. It was auto checked out "esp-idf-v4.3" so I switched to "public" and updated Moddable SDK as you said. 
Also the installation of ESP IDF didn't clone correctly the repository, so you were in fact right in both points 😅

Now it's working, thank you Andy!

**Reply by andycarle** [reply]
Happy to help! I'm glad to hear that you've got everything working now.

I've also deleted that old `esp-idf-v4.3` branch to avoid future confusion. Thank you for pointing that out!
